README.md

```diff
@@ -1,52 +1,167 @@
 
-ajax communicates with database without need for page refresh - helps with single page applications
+Postman:
+https://api.github.com/repos/rails/rails/commits?sha=82885325e04d78fb7ec608a4670164d842d23078
+Returns success
 
-use AJAX `get` method to retrieve some text from `index/sentence.html`  and display the text in index.html
+Postman error:
+https://api.github.com/repos/rails/rails/commits?sha=fake-SHA
+Returns a 404 not found
 
-use `python -m SimpleHTTPServer` to serve page to make request
+A good developer will make sure to handle these unexpected events gracefully when using Ajax. We can provide multiple callbacks when using jQuery: one to handle a successful response and one to handle when an error occurs.
 
- 
-## Callbacks
+```javascript
+$.get("this_doesnt_exist.html", function(data) {
+  // This will not be called because the .html file request doesn't exist
+})
+.fail(function(error) {
+  // This is called when an error occurs
+  console.log('Something went wrong: ' + error);
+});
+```
 
-ajax follows an asynchronous pattern, the get and post requests executes the success code block only when it gets a response.
+We chained an additional call to ```fail``` on the end of our request. We passed a function to the method which will run only if an error occurs. In our example we logged the error to the console, but in real world situation you might want to try to fix the issue or inform the user.
 
-it knows it will take a while to GET the data, so while it's doing it, it will continue on with the rest of the code, and then execute the success block WHEN it's successful.
+This is another example of how we could use jQuery to perform an Ajax request.
 
-we tell it what to get, and then we tell it what to do once it's finished. 
+**without passing it an argumnet- gets it automagically**
+```javascript
+var url = "https://api.github.com/repos/rails/rails/commits?sha=82885325e04d78fb7ec608a4670164d842d23078";
 
-like putting food in the microwave, you personally don't stand and wait until it's finished heating your food, you probably play on your phone, look at instagram, maybe watch some TV or do more work on learn. you go on and do other things until your food is heated, and then you remove it from the microwave and eat it (the success function)
+$.get(url)
+  .done(function(data) {
+    console.log("Done");
+    console.log(data);
+  });
 
-if error - error callbacks
+```
 
-show more examples with code after - console.log
+Note: The callback that gets passed into `.done`  gets `data` as an argument.  `data` represents the response returned from the API. jQuery handles passing in that `data` object to the callbacks.  This is essential to our fire and forget technique.  We don't have to sit around and wait for the API to give us a response.  Instead, we tell jQuery that when it receives a response to please pass it along to our callbacks so they can handle it accordingly.
```

Comment:
This last paragraph could use a bit more explication. If we’re going to use jQuery for AJAX, do we want to explore how it handles different content types? Which reminds me, Ajax has “XML” in it, but this Readme doesn’t address how JavaScript might handle XML (or the fact that almost everyone uses JSON or Transit or anything but XML these days). Should we dive into that topic (in a shallow way) above?